packages/core/lib/commands/call/run.js

```diff
@@ -0,0 +1,218 @@
+module.exports = async function (options) {
+  const fs = require("fs");
+  const util = require("util");
+  const { Environment } = require("@truffle/environment");
+  const OS = require("os");
+  const Codec = require("@truffle/codec");
+  const Encoder = require("@truffle/encoder");
+  const Decoder = require("@truffle/decoder");
+  const TruffleError = require("@truffle/error");
+  const { fetchAndCompile } = require("@truffle/fetch-and-compile");
+  const loadConfig = require("../../loadConfig");
+  const DebugUtils = require("@truffle/debug-utils");
+  const web3Utils = require("web3-utils");
+  const { getFirstDefinedValue } = require("../../configAdapter");
+  const { formatBlockSpecifier } = require("../../../../encoder/dist/adapter");
+
+  if (options.url && options.network) {
+    const message =
+      "" +
+      "Mutually exclusive options, --url and --network detected!" +
+      OS.EOL +
+      "Please use either --url or --network and try again." +
+      OS.EOL +
+      "See: https://trufflesuite.com/docs/truffle/reference/truffle-commands/#call" +
+      OS.EOL;
+    throw new TruffleError(message);
+  }
+
+  let config = loadConfig(options);
+  await Environment.detect(config);
+
+  const [contractNameOrAddress, functionNameOrSignature, ...args] = config._;
+  let functionEntry, transaction;
+
+  // Validates the block number and returns the hex string representation of a number or
+  // an error if the string is other than "latest", "pending" and "earliest" or
+  // a hex string representation of a number
+  const blockToFetch = formatBlockSpecifier(config.blockNumber);
```

Comment:
Well, it *throws* an error if it's not formatted right, right?  But such an error is an expected error, so it should probably get turned into a `TruffleError`.  Maybe that's not so important, idk.